ICGE-Ui/src/main/java/de/unistuttgart/informatik/fius/icge/ui/internal/SwingEntityInspector.java

```
@@ -93,30 +102,130 @@ public void setName(final String name) {
+                this.repaint();
+            }
+        });
+    }
+    
+    /**
+     * Check if the new list of entries matches the current entries structurally.
+     *
+     * @param entries
+     *     The ne list of entries to check against currentEntries
+     * @return true iff only the values need to be updated
+     */
+    private boolean canUpdateValuesOnly(final EntityInspectorEntry[] entries) {
+        if (this.currentEntries == null) return false;
+        if (this.currentEntries.length != entries.length) return false;
+        for (int i = 0; i < entries.length; i++) {
+            EntityInspectorEntry oldEntry = this.currentEntries[i];
+            EntityInspectorEntry newEntry = entries[i];
+            if (!oldEntry.getName().equals(newEntry.getName())) return false;
+            if (!oldEntry.getType().equals(newEntry.getType())) return false;
+            // only value can differ for updates without complete redraws!
+        }
+        return true;
+    }
+    
+    /**
+     * Callback to change the value of an entry.
+     * 
+     * This indirect callback is used to avoid changing the callback on the ui element when only the values of the
+     * entries have changed.
      * 
+     * @param index
+     *     the index of the entry to update
      * @param name
-     *     The name of the setting, which is displayed infromt
-     * @param type
-     *     The type of the setting
+     *     the name of the entry to update (additional safeguard)
      * @param value
-     *     The default/start value of the element
-     * @param callback
-     *     The callback run on value change
+     *     the new value to set for the entry
      */
-    public void addUIElement(final String name, final String type, final String value, final BiConsumer<String, String> callback) {
+    private void updateValueOnEntity(int index, String name, String value) {
+        if (this.currentEntries == null || index >= this.currentEntries.length) return;
```

Maybe throw an exception? Same as above.

---

If any of these does not match then the ui was n the middle of updating. Then it should be ok to silently dismiss the request. The user will see the ui updating and should be able to tell why the request did not work. It should also be nigh impossible to trigger this condition manually (only selecting a new entity for inspection can cause these conditions).